keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -2840,143 +2876,271 @@ GA:
+`apiserver_flowcontrol_request_queue_length_after_enqueue` metric
+getting too close to the configured queue length could be a sign of
+insufficient queue size (or a system overload), which can be precursor
+to rejected requests.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Manual tests were run during promotion to Beta in 1.20.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+The rollout replaces the behavior of the max-in-flight filter with the
+more sophisticated behavior of this feature.
 
 ### Monitoring Requirements
 
-* **How can an operator determine if the feature is in use by workloads?**
-  If the `apiserver_flowcontrol_dispatched_requests_total` metric is non-zero,
-  this feature is in use. Note that this isn't a workload feature, but a
-  control plane one.
+###### How can an operator determine if the feature is in use by workloads?
+
+If the `apiserver_flowcontrol_dispatched_requests_total` metric is
+non-zero, this feature is in use. Note that this isn't a workload
+feature, but a control plane one.
+
+###### How can someone using this feature know that it is working for their instance?
+
+The classification functionality can be reviewed by looking at the
+`apf_pl` and `apf_fs` attributes that appear in the httplog lines from
+the apiserver when `-v=3` or more verbose.
+
+The nominal division of concurrency can be checked by looking at the
+`apiserver_flowcontrol_nominal_limit_seats` metric of the limited
+priority levels to see whether they are in the proportions given by
+the `NominalConcurrencyShares` fields in the corresponding
+`PriorityLevelConfiguration` objects, and sum to the server's
+concurrency limit (the sum of the `--max-requests-inflight` and
+`--max-mutating-requests-inflight` command line parameters) allowing
+for rounding error.
+
+The dynamic adjustment of concurrency limits can be checked for
+validity by checking whether the
+`apiserver_flowcontrol_current_limit_seats` metrics sum to the
+server's concurrency limit allowing for rounding error.  The
+responsiveness to load can be checked by comparing those dynamic
+concurrency limits to various indications of load and keeping in mind
+the bounds shown by the `apiserver_flowcontrol_lower_limit_seats` and
+`apiserver_flowcontrol_upper_limit_seats` metrics.  Indications of
+load include the `apiserver_flowcontrol_request_wait_duration_seconds`
+histogram vector metric and the
+`apiserver_flowcontrol_demand_seats_smoothed` gauge vector metric.
+
+The dispatching can be checked by checking whether there are
+significant queuing delays only for flow schemeas that go to priority
+levels whose seats are highly utilized.  The histogram vector metric
+`apiserver_flowcontrol_request_wait_duration_seconds` shows the
+former, and tThe histogram vector metric
+`apiserver_flowcontrol_priority_level_seat_utilization` shows the
+latter.
```

Comment:
I'd also mention that responses to requests include a header indicating the FSC and PL the request matched, and that can be used to ensure that requests are going to the right places?

Reply:
Done in the force-push to 32bb772a